spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/OutputBindingLifecycle.java

```diff
@@ -61,8 +62,10 @@ public void start() {
 				throw new IllegalStateException(
 						"Cannot perform binding, no proper implementation found", e);
 			}
-			this.running = true;
-			this.applicationContext.start();
+			if (channelBindingServiceProperties != null && channelBindingServiceProperties.isAutoStartup()) {
```

Comment:
Same as https://github.com/spring-cloud/spring-cloud-stream/pull/527/files#r62900470.

Only the context autostart may be optional.